lib/trusted-mpt-light-client-types/src/signed_data.rs

```diff
@@ -0,0 +1,45 @@
+use unionlabs::{
+    encoding::{Encode, Encoding},
+    primitives::{H256, H512},
+};
+
+#[derive(Debug, Clone, PartialEq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "bincode", derive(bincode::Encode, bincode::Decode))]
+pub struct SignedData<T> {
+    data: T,
+    signature: H512,
```

Comment:
another small nit, let's make an issue for it to be done later: this type could be generic over the signature size as well. we could then look into moving this type into somewhere more generic. for now this is fine though

Reply:
https://github.com/unionlabs/union/issues/4207